icinga2/icinga.go

```diff
@@ -141,6 +161,40 @@ func (s *WebClient) FilteredQuery(url string, filter QueryFilter, result, errmsg
 	return s.napping.Send(&req)
 }
 
+func (s *WebClient) SetIcingaUrl(url string) {
+	s.URL = url
+}
+
+func (s *MockClient) SetIcingaUrl(url string) {
+	conf := s.GetClientConfig()
+	conf.SetIcingaUrl(url)
+}
+
+func (s *WebClient) TestIcingaApi(url string) error {
```

Comment:
Having this method take a separate `url` as a parameter means that it doesn't really need to be a method on the Icinga client at all.

In my opinion, having this method not take an argument and instead testing that the URL of the client on which it is called is working would make more sense.